packages/eslint-plugin/src/rules/no-base-to-string.ts

```diff
@@ -199,7 +206,8 @@ export default createRule<Options, MessageIds>({
       if (
         declarations.every(
           ({ parent }) =>
-            !ts.isInterfaceDeclaration(parent) || parent.name.text !== 'Object',
+            !ts.isInterfaceDeclaration(parent) ||
+            (parent.name.text !== 'Object' && parent.name.text !== 'Array'),
```

Comment:
This is necessary so unions/intersections that include an array type will be matched as unions/intersections.